vms/avm/tx_builders.go

```diff
@@ -0,0 +1,229 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package avm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/set"
+	"github.com/ava-labs/avalanchego/vms/avm/txs"
+	"github.com/ava-labs/avalanchego/vms/components/avax"
+	"github.com/ava-labs/avalanchego/vms/components/verify"
+	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
+	"github.com/ava-labs/avalanchego/wallet/chain/x/signer"
+	"github.com/ava-labs/avalanchego/wallet/subnet/primary/common"
+
+	walletbuilder "github.com/ava-labs/avalanchego/wallet/chain/x/builder"
+)
+
+type txBuilderBackend interface {
+	walletbuilder.Backend
+	signer.Backend
+
+	Context() *walletbuilder.Context
+	ResetAddresses(addrs set.Set[ids.ShortID])
+}
```

Comment:
What do we need this interface for? Wondering if we need this

Reply:
We don't need context, but we need the whole interface to allow wallet builders and signers to retrieve UTXOs and sign transactions.
I could not find a way to avoid the ResetAddresses, as we managed in `platformvm` because we have two different backends here